addons/launcher-addon/src/main/java/io/fabric8/launcher/addon/preparers/InsertMavenDependenciesPreparer.java

```diff
@@ -0,0 +1,80 @@
+package io.fabric8.launcher.addon.preparers;
+
+import java.nio.file.Path;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.io.StringReader;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.json.Json;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+import javax.json.JsonValue;
+import javax.json.stream.JsonParsingException;
+
+import io.fabric8.launcher.booster.catalog.rhoar.RhoarBooster;
+import io.fabric8.launcher.core.api.CreateProjectileContext;
+import io.fabric8.launcher.core.api.ProjectileContext;
+import io.fabric8.launcher.core.spi.ProjectilePreparer;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.Dependency;
+import org.jboss.forge.addon.maven.resources.MavenModelResource;
+import org.jboss.forge.addon.resource.DirectoryResource;
+import org.jboss.forge.addon.resource.ResourceFactory;
+
+
+@ApplicationScoped
+public class InsertMavenDependenciesPreparer implements ProjectilePreparer {
+    @Inject
+    private ResourceFactory resourceFactory;
+
+    @Override
+    public void prepare(Path projectPath, RhoarBooster booster, ProjectileContext context) {
+        if (!(context instanceof CreateProjectileContext)) {
+            return;
+        }
+        Logger LOG = Logger.getLogger(InsertMavenDependenciesPreparer.class.getName());
+        CreateProjectileContext createProjectileContext = (CreateProjectileContext) context;
+        DirectoryResource projectDirectory = resourceFactory.create(projectPath.toFile()).as(DirectoryResource.class);
+        MavenModelResource modelResource = projectDirectory.getChildOfType(MavenModelResource.class, "pom.xml");
+
+        // Perform model changes
+        if (modelResource.exists()) {
+            Model model = modelResource.getCurrentModel();
+            String dependencySnapshot = createProjectileContext.getDependencySnapshot();
+            JsonReader reader = Json.createReader(new StringReader(dependencySnapshot));
```

Comment:
Put in the try-finally block instead. Then you don't need to explicitly close it